src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,9 +1,73 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
+
+    public Object getInstance(Class<?> interfaceClass) {
+        Object interfaceClassInstance = null;
+        Class<?> interfaceImplClass = findImplementation(interfaceClass);
+        Field[] fields = interfaceImplClass.getDeclaredFields();
+        for (Field field : fields) {
+            if (field.isAnnotationPresent(Component.class)) {
+                Object fieldInstance = getInstance(field.getType());
+                interfaceClassInstance = createNewInstance(interfaceImplClass);
+                try {
+                    field.setAccessible(true);
+                    field.set(interfaceClassInstance, fieldInstance);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("Can't initialize field "
+                            + field.getName() + " in Class "
+                            + interfaceImplClass.getName(), e);
+                }
+            }
+        }
+        return interfaceClassInstance == null ? createNewInstance(interfaceImplClass)
+                : interfaceClassInstance;
+    }
+
+    private Object createNewInstance(Class<?> interfaceClassImpl) {
+        if (!interfaceClassImpl.isAnnotationPresent(Component.class)) {
```

Comment:
let's move this check to findImplementation method.

Reply:
done